gateway/conf.d/apicast.conf

```diff
@@ -76,71 +76,57 @@ location @upstream {
     require('resty.ctx').apply()
   }
 
-  #{% capture proxy_cache_valid %}
-  #{#} proxy_cache $cache_zone;
-  #{#} proxy_cache_key $scheme$request_method$proxy_host$request_uri$service_id;
-  #{#} proxy_no_cache $cache_request;
-  #{#} proxy_cache_valid {{ env.APICAST_CACHE_STATUS_CODES | default: '200 302'}} {{ env.APICAST_CACHE_MAX_TIME | default: '1m' }};
-  #{% endcapture %}
-  #{{ proxy_cache_valid | replace: "#{#}", "" }}
-  #
-
-  #{% if opentelemetry != empty %}
-  #   {% capture opentelemetry_propagate_directive %}
-  #{#} opentelemetry_propagate;
-  #   {% endcapture %}
-  #   {{ opentelemetry_propagate_directive | replace: "#{#}", "" }}
-  #{% endif %}
+  {% include "conf.d/upstream_shared.conf" %}
 
-  proxy_pass $proxy_pass;
+  # these are duplicated so when request is redirected here those phases are executed
+  post_action @out_of_band_authrep_action;
+  body_filter_by_lua_block { require('apicast.executor'):body_filter() }
+  header_filter_by_lua_block { require('apicast.executor'):header_filter() }
+}
 
-  proxy_http_version 1.1;
-  proxy_set_header X-Real-IP  $remote_addr;
-  proxy_set_header Host $http_host;
-  proxy_set_header X-3scale-proxy-secret-token $secret_token;
-  proxy_set_header X-3scale-debug "";
-  proxy_set_header Connection $upstream_connection_header;
-  proxy_set_header Upgrade $upstream_upgrade_header;
-
-  # This is a bit tricky. It uses liquid to set a SSL client certificate. In
-  # NGINX, all this is not executed as it is commented with '#'. However, in
-  # Liquid, all this will be evaluated. As a result, the following directives
-  # are set optionally: proxy_ssl_certificate, proxy_ssl_certificate_key,
-  # proxy_ssl_session_reuse, and proxy_ssl_password_file.
-
-  # {% if proxy_ssl_certificate != empty and proxy_ssl_certificate_key != empty %}
-  #   {% capture proxy_ssl %}
-  #{#}   proxy_ssl_certificate {{ proxy_ssl_certificate }};
-  #{#}   proxy_ssl_certificate_key {{ proxy_ssl_certificate_key }};
-  #   {% endcapture %}
-  #   {{ proxy_ssl | replace: "#{#}", "" }}
-  #
-  #   {% if proxy_ssl_password_file != empty %}
-  #     {% capture proxy_ssl %}
-  #{#}   proxy_ssl_password_file {{ proxy_ssl_password_file }};
-  #     {% endcapture %}
-  #   {{ proxy_ssl | replace: "#{#}", "" }}
-  #   {% endif %}
-  #
-  #   {% if proxy_ssl_session_reuse != empty %}
-  #     {% capture proxy_ssl %}
-  #{#}   proxy_ssl_session_reuse {{ proxy_ssl_session_reuse }};
-  #     {% endcapture %}
-  #   {{ proxy_ssl | replace: "#{#}", "" }}
-  #   {% endif %}
-  # {% endif %}
+location @upstream_unbuffered {
+  internal;
+
+  rewrite_by_lua_block {
+    require('resty.ctx').apply()
+  }
+
+  proxy_request_buffering off;
+  proxy_buffering off;
```

Comment:
disabling (response) buffering can affect caching. We should analyze the impact.

Reply:
The response buffering is removed from the policy